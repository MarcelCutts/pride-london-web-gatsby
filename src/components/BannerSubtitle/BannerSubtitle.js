import React from 'react'
import styled from 'styled-components'

const BannerSubtitle = styled.h3`
  color: #ffffff;
  font-size: 16px;

  & span {
    background-color: #2d2f7f;
    display: inline-block;
    padding: 0 10px;
  }
`
const StyledBannerSubtitle = ({ children }) => (
  <BannerSubtitle>
    <span>{children}</span>
  </BannerSubtitle>
)

export default StyledBannerSubtitle
