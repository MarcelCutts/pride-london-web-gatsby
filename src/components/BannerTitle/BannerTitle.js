import React from 'react'
import styled from 'styled-components'

const BannerTitle = styled.h1`
  display: block;
  color: #2d2f7f;
  font-size: 28px;
  margin-bottom: 0;

  & span {
    background-color: #ffffff;
    display: inline-block;
    padding: 0 10px;
  }
`

const StyledBannerTitle = ({ children }) => (
  <BannerTitle>
    <span>{children}</span>
  </BannerTitle>
)

export default StyledBannerTitle
