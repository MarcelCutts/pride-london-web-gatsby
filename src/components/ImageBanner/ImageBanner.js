import React from 'react'

const ImageBanner = ({ src }) => <div />

export default ImageBanner
