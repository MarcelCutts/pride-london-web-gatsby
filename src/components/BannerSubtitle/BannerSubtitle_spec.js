import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { shallow } from 'enzyme'

import BannerSubtitle from '../BannerSubtitle/BannerSubtitle'

describe('BannerSubtitle', () => {
  it('should render', () => {
    const tree = renderer.create(<BannerSubtitle />)
    expect(tree).toMatchSnapshot()
  })

  it('should render a <span />', () => {
    const wrapper = shallow(<BannerSubtitle />)
    expect(wrapper.find('span')).toHaveLength(1)
  })

  it('should render child text ', () => {
    const text = 'And here is a test subtitle!'
    const wrapper = shallow(<BannerSubtitle>{text}</BannerSubtitle>)
    expect(wrapper.find('span').text()).toBe(text)
  })
})
