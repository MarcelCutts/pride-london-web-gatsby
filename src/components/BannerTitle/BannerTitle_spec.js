import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { shallow } from 'enzyme'

import BannerTitle from '../BannerTitle/BannerTitle'

describe('BannerTitle', () => {
  it('should render', () => {
    const tree = renderer.create(<BannerTitle />)
    expect(tree).toMatchSnapshot()
  })

  it('should render a <span />', () => {
    const wrapper = shallow(<BannerTitle />)
    expect(wrapper.find('span')).toHaveLength(1)
  })

  it('should render child text ', () => {
    const text = 'Here is a test title!'
    const wrapper = shallow(<BannerTitle>{text}</BannerTitle>)
    expect(wrapper.find('span').text()).toBe(text)
  })
})
