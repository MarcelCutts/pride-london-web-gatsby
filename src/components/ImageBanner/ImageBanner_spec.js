import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { shallow, mount } from 'enzyme'

import ImageBanner, { BackgroundImage } from './ImageBanner'
import BannerTitle from '../BannerTitle/BannerTitle'
import BannerSubtitle from '../BannerSubtitle/BannerSubtitle'

describe('ImageBanner', () => {
  it('should render', () => {
    const wrapper = shallow(<ImageBanner />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render a <BannerTitle />', () => {
    const wrapper = shallow(<ImageBanner />)
    expect(wrapper.find(BannerTitle)).toHaveLength(1)
  })

  it('should render a <BannerSubtitle />', () => {
    const wrapper = shallow(<ImageBanner />)
    expect(wrapper.find(BannerSubtitle)).toHaveLength(1)
  })

  it('should render the titleText from props to BannerTitle ', () => {
    const titleText = 'Here is a test title!'
    const wrapper = mount(<ImageBanner titleText={titleText} />)
    expect(wrapper.find(BannerTitle).text()).toBe(titleText)
  })

  it('should render the subtitleText from props to BannerTitle ', () => {
    const subtitleText = 'And here is a test subtitle!'
    const wrapper = mount(<ImageBanner subtitleText={subtitleText} />)
    expect(wrapper.find(BannerSubtitle).text()).toBe(subtitleText)
  })

  it('should pass the imageSrc prop to the BackgroundImage component', () => {
    const imageSrc = 'http://www.pics.com/coolpic'
    const wrapper = shallow(<ImageBanner imageSrc={imageSrc} />)
    expect(wrapper.find(BackgroundImage).props().imageSrc).toBe(imageSrc)
  })
})
