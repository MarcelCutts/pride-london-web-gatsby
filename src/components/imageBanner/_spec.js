import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { shallow } from 'enzyme'

import ImageBanner, { BackgroundImage } from './'
import BannerTitle from '../bannerTitle'
import BannerSubtitle from '../bannerSubtitle'

describe('ImageBanner', () => {
  it('should render', () => {
    const tree = renderer.create(<ImageBanner />)
    expect(tree).toMatchSnapshot()
  })

  it('should render a <BannerTitle />', () => {
    const wrapper = shallow(<ImageBanner />)
    expect(wrapper.find(BannerTitle)).toHaveLength(1)
  })

  it('should render a <BannerSubtitle />', () => {
    const wrapper = shallow(<ImageBanner />)
    expect(wrapper.find(BannerSubtitle)).toHaveLength(1)
  })

  it('should render an <img> if imageSrc is passed', () => {
    const wrapper = shallow(<ImageBanner imageSrc="test" />)
    expect(wrapper.find('img')).toHaveLength(1)
  })

  it('should not render an <img> if no imageSrc is passed', () => {
    const wrapper = shallow(<ImageBanner />)
    expect(wrapper.find('img')).toHaveLength(0)
  })

  it('should render an img with imgSrc from props', () => {
    const imageSrc = 'http://www.image.com'
    const wrapper = shallow(<ImageBanner imageSrc={imageSrc} />)
    expect(wrapper.find('img').props().src).toBe(imageSrc)
  })

  it('should render an img with altText from props', () => {
    const altText = 'background image'
    const wrapper = shallow(<ImageBanner imageSrc="test" altText={altText} />)
    expect(wrapper.find('img').props().alt).toBe(altText)
  })

  it('should render the titleText from props to BannerTitle ', () => {
    const titleText = 'Here is a test title!'
    const wrapper = shallow(<ImageBanner titleText={titleText} />)
    expect(
      wrapper
        .find(BannerTitle)
        .dive()
        .find('span')
        .text()
    ).toBe(titleText)
  })

  it('should render the subtitleText from props to BannerTitle ', () => {
    const subtitleText = 'And here is a test subtitle!'
    const wrapper = shallow(<ImageBanner subtitleText={subtitleText} />)
    expect(
      wrapper
        .find(BannerSubtitle)
        .dive()
        .find('span')
        .text()
    ).toBe(subtitleText)
  })
})
