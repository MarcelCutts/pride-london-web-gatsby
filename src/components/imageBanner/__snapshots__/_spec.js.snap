// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageBanner should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ImageBanner
    altText=""
    color=""
    imageSrc=""
    subtitleText=""
    titleText=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        "",
        <div>
          <StyledBannerTitle>
            
          </StyledBannerTitle>
          <StyledBannerSubtitle>
            
          </StyledBannerSubtitle>
        </div>,
      ],
      "color": "",
    },
    "ref": null,
    "rendered": Array [
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <StyledBannerTitle>
              
            </StyledBannerTitle>,
            <StyledBannerSubtitle>
              
            </StyledBannerSubtitle>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          "",
          <div>
            <StyledBannerTitle>
              
            </StyledBannerTitle>
            <StyledBannerSubtitle>
              
            </StyledBannerSubtitle>
          </div>,
        ],
        "color": "",
      },
      "ref": null,
      "rendered": Array [
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <StyledBannerTitle>
                
              </StyledBannerTitle>,
              <StyledBannerSubtitle>
                
              </StyledBannerSubtitle>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ImageBanner should render the subtitleText from props to BannerTitle  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ImageBanner
    altText=""
    color=""
    imageSrc=""
    subtitleText="And here is a test subtitle!"
    titleText=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        "",
        <div>
          <StyledBannerTitle>
            
          </StyledBannerTitle>
          <StyledBannerSubtitle>
            And here is a test subtitle!
          </StyledBannerSubtitle>
        </div>,
      ],
      "color": "",
    },
    "ref": null,
    "rendered": Array [
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <StyledBannerTitle>
              
            </StyledBannerTitle>,
            <StyledBannerSubtitle>
              And here is a test subtitle!
            </StyledBannerSubtitle>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "And here is a test subtitle!",
            },
            "ref": null,
            "rendered": "And here is a test subtitle!",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          "",
          <div>
            <StyledBannerTitle>
              
            </StyledBannerTitle>
            <StyledBannerSubtitle>
              And here is a test subtitle!
            </StyledBannerSubtitle>
          </div>,
        ],
        "color": "",
      },
      "ref": null,
      "rendered": Array [
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <StyledBannerTitle>
                
              </StyledBannerTitle>,
              <StyledBannerSubtitle>
                And here is a test subtitle!
              </StyledBannerSubtitle>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "And here is a test subtitle!",
              },
              "ref": null,
              "rendered": "And here is a test subtitle!",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ImageBanner should render the titleText from props to BannerTitle  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ImageBanner
    altText=""
    color=""
    imageSrc=""
    subtitleText=""
    titleText="Here is a test title!"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        "",
        <div>
          <StyledBannerTitle>
            Here is a test title!
          </StyledBannerTitle>
          <StyledBannerSubtitle>
            
          </StyledBannerSubtitle>
        </div>,
      ],
      "color": "",
    },
    "ref": null,
    "rendered": Array [
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <StyledBannerTitle>
              Here is a test title!
            </StyledBannerTitle>,
            <StyledBannerSubtitle>
              
            </StyledBannerSubtitle>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Here is a test title!",
            },
            "ref": null,
            "rendered": "Here is a test title!",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          "",
          <div>
            <StyledBannerTitle>
              Here is a test title!
            </StyledBannerTitle>
            <StyledBannerSubtitle>
              
            </StyledBannerSubtitle>
          </div>,
        ],
        "color": "",
      },
      "ref": null,
      "rendered": Array [
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <StyledBannerTitle>
                Here is a test title!
              </StyledBannerTitle>,
              <StyledBannerSubtitle>
                
              </StyledBannerSubtitle>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Here is a test title!",
              },
              "ref": null,
              "rendered": "Here is a test title!",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
